src/app/modules/vault-filter/vault-filter.component.ts

```diff
@@ -18,17 +18,11 @@ export class VaultFilterComponent extends BaseVaultFilterComponent {
   searchPlaceholder: string;
   searchText = "";
 
-  organization: Organization;
-
-  constructor(vaultFilterService: VaultFilterService) {
+  constructor(protected vaultFilterService: VaultFilterService) {
```

Comment:
Thoughts on https://github.com/bitwarden/jslib/pull/822 and [[refactor] Establish pattern for interfaces in module scoped services](https://github.com/bitwarden/web/pull/1700/commits/c808aa8a15be1ec121754d565ab6aac910dbbe74)?

Pros:
- Removes the constructor in `web/VaultFilterComponent`, uses DI to inject the right service from an abstraction
- Still allows for extending module services and their abstractions in clients for stuff that doesn't belong in jslib so we can override methods or make new ones

Cons:
- We have to do the weird thing where we provide the right service for an extended interface and its base interface
- We still have to have a special constructor in `web/OrganizationVaultFiltersComponent` so it knows about the web specific `buildAdminCollections` method. I think this is less confusing, since if you remove the constructor you will immediately get a syntax error.

Reply:
Marking this as resolved because it's out of scope for this PR